string name
string[] wp_list
